gnucash/report/trep-engine.scm

```diff
@@ -2236,114 +2215,146 @@ be excluded from periodic reporting.")
+                      (eq? reconcile-filter 'all)
+                      (eq? void-filter 'non-void-only)
+                      (memq primary-key '(account-name account-code))
+                      (memq secondary-key '(register-order date))
+                      primary-subtotal
+                      (eq? secondary-order 'ascend)))
+              )))
+
+        ;; append additional parameters that rely on sortkey-list and therefore
+        ;; whose definition require an existing parameters alist
+        (append parameters
+          (let* ((primary-get-info (lambda (info)
+                                    (if (memq primary-key DATE-SORTING-TYPES)
+                                        (keylist-get-info date-subtotal-list primary-date-subtotal info)
+                                        (and (SUBTOTAL-ENABLED? primary-key parameters)
+                                              primary-subtotal
+                                              (keylist-get-info (sortkey-list parameters) primary-key info)))))
+                (primary-key/renderer-fn (primary-get-info 'renderer-fn))
+                (secondary-get-info (lambda (info)
+                                      (if (memq secondary-key DATE-SORTING-TYPES)
+                                          (keylist-get-info date-subtotal-list secondary-date-subtotal info)
+                                          (and (SUBTOTAL-ENABLED? secondary-key parameters)
+                                              secondary-subtotal
+                                              (keylist-get-info (sortkey-list parameters) secondary-key info)))))
+                (secondary-key/renderer-fn (secondary-get-info 'renderer-fn)))
+            (list
+              ;; additional parameters based on display options
+              (cons 'running-prime
+                    (and primary-key/renderer-fn
+                         (or (eq? (opt-val gnc:pagename-display optname-running-totals) 'sub)
+                             (eq? (opt-val gnc:pagename-display optname-running-totals) 'all))))
+              (cons 'running-sec
+                    (and secondary-key/renderer-fn
+                         (or (eq? (opt-val gnc:pagename-display optname-running-totals) 'sub)
+                             (eq? (opt-val gnc:pagename-display optname-running-totals) 'all))))
+              ;; additional parameters based on sorting options
+              (cons 'primary-key/renderer-fn primary-key/renderer-fn)
+              (cons 'primary-key/split-sortvalue (primary-get-info 'split-sortvalue))
+              (cons 'secondary-key/renderer-fn secondary-key/renderer-fn)
+              (cons 'secondary-key/split-sortvalue (secondary-get-info 'split-sortvalue))
+              (cons 'subtotals-only
+                    (and (opt-val pagename-sorting optname-show-subtotals-only)
+                          (or primary-key/renderer-fn secondary-key/renderer-fn)))
+            )))))
```

Comment:
Instead of using `parameters` as both the shorter and expanded varname, I'd name the shorter `parameters-tail` and the expanded one `parameters` and use the following construct instead of `append`:

```scheme
(let ((parameters-tail (list ...)))
  (let ((primary-get-info ...))
    (cons* (cons 'running-prime ...)
           (cons 'running-sec ...)
           parameters-tail)))
```

It's still a two-step process but is tidier.

Reply:
Interesting. I originally looked at list* but then couldn't find it in guile. Didn't realize it's cons*. Now I see srfi-1 says

> This function is called list* in [Common Lisp](https://srfi.schemers.org/srfi-1/srfi-1.html#CommonLisp) and about half of the Schemes that provide it, and cons* in the other half.

Reply:
updated as per recommendation